### ISSUE TITLE
Visit History's "Results per page" applies only to encounters, not documents

### ISSUE THREAD

**Issue by pete-boyd** [openemr/openemr#7275]
In OpenEMR 7.0.1 (1), in Visit History, if you set "Results per page", it only affects encounters. Any documents uploaded to the patient record are not included in the total figure defined by "Results per page", and appear at the end of the list of encounters.

E.g., set "Results per page: 5" and 5 encounters are listed, but then 8 of "Document:...".

**Comment by sjpadgett** [comment]
@pete-boyd Yeah I can see how this would be annoying especially for paperless clinics that upload most everything dealing with a patient.
Maybe another range for documents included like the last 5 or latest. Or only docs attached to an encounter. I don't know how much attaching documents to encounters is used in the wild and would need feedback for that.
What would be your recommendation Pete?

**Comment by pete-boyd** [comment]
Here's a description of how Visit History is being used:

"Some of the GP's that run our clinic like to see the full history from the EMR for each client (they do not have access to the EMR but our RGN running the clinic does). For this reason we routinely print the list of encounters for each client and that is given to the GP when the client arrives for appointment. So a clinic might have 10 appointments we print 10 pages, the client name and DOB and list of encounters."

In this scenario, I'm imagining that any info on documents is superfluous.

I've made an inquiry with regard to your questions though, and will get back to you.

**Comment by ruthkonyn** [comment]
I think there are two issues:
1. the way the results per page count works is buggy - it reports say N results but displays N+3 entries, where there are 3 documents and N encounters. In some circumstances, such as the 'ALL' setting for results per page the only problem is the reported results count (documents are listed in date order amongst the encounters) , but for others, e.g. '5' or '20' there may be extra documents listed on the last page as well as those same documents included with the list of encounters, i.e. in date order. I'm looking at how to fix this bug so that documents are included in the results counts, and are listed in date order in the history, not at the end (or is the intended behaviour only to list non encounter attached documents at the end, not also in date order in the history?)
2. the other is whether a user would want documents listed at all, in this case enhance with another set of document options: 'do not include any documents' 'include only documents attached to an encounter' 'include all documents'

**Comment by pete-boyd** [comment]
I received this in relation to documents appearing in Visit History:
"Actually useful for us to see, in visit history, if documents have been attached. For example when an existing/previously seen client presents to triage we check the visit history in the EMR to see if there is a Document xxxxx.pdf (patient id card) and use this as a check in our process."
